simavr/sim/sim_gdb.c

```diff
@@ -655,6 +655,10 @@ avr_gdb_init(
 	avr->sleep = avr_callback_sleep_gdb;
 	
 	return 0;
+
+	error:
+		free(g);
```

Comment:
Yet another thing..
I am writing GUI for simavr where it's possible start and stop and then start again, repeatably,  the debug session. In these conditions, the avr_deinit_gdb function also lead to SEGFAULT, so I had to rewrite it this way:
void 
avr_deinit_gdb(
		avr_t * avr )
{
    if (!avr->gdb)
        return;
    avr->run = avr_callback_run_raw; // restore normal callbacks
    avr->sleep = avr_callback_sleep_raw;
    if (avr->gdb->listen != -1)
	   close(avr->gdb->listen);
    avr->gdb->listen = -1;
	if (avr->gdb->s != -1)
	   close(avr->gdb->s);
    avr->gdb->s = -1;
	free(avr->gdb);
    avr->gdb = NULL;

	network_release();
}